data-reset-tool/Refactor.md

```diff
@@ -0,0 +1,54 @@
+
+## Objective
+
+The goal is to decouple the logic in the data reset tool from the models in legal API. This will result in a simpler, easier to maintain tool with no dependencies on the code in legal-api. We do not want to change the endpoints or the reset tool's API structure too much to reduce the amount of refactoring that has to be done in the nightwatch scripts.   
+
+## Motivation
+
+The models and the database structure in LEAR change frequently. If a change goes in without being reflected in the data reset tool, there could either be missed data or catastrophic failures. Decoupling the data reset tool from the models in legal-api would result in more reliability and predicatability within the data reset tool. By using postgres' built in bulk copy methods we should also see a performance improvement over the current design.   
+
+## Design Proposal
+
+The proposal is to gut the dependencies on the legal-api models from the current data-reset tool code and replace them with native postgres bulk import and export mechanisms (copy and copy_expert). Postgres has the ability to map columnar data in CSV format to tables. This would change the current code structure in the following ways:
+
+* The JSONConverter class would either no longer be needed, or would be changed to not call methods of the entities. A more agnostic approach may be to use the GET methods in legal API to retrieve entity data.
+
+* The folder structure where the test data is held (e2e/lear-data) would change to have a folder per corp num. Each folder would have a csv file per table (ie CP0000001/directors.csv) since bulk copy cannot work with multiple tables.
```

Comment:
Is there anyway just to restore a group of businesses from the back up avoiding the pain of managing and maintaining the  CSV files

Reply:
From a db backup? Of course, but that would remove the ability to set custom scenarios. Or did you mean reduce the number of CSV files? I.e - have multiple businesses in a business.csv instead of a business.csv per corp